src/async_reader.rs

```diff
@@ -141,15 +155,25 @@ impl<B: AsyncBackend + Sync + Send> AsyncPmTilesReader<B> {
     }
 
     #[async_recursion]
-    async fn find_entry_rec(&self, tile_id: u64, entry: &Entry, depth: u8) -> Option<Entry> {
+    async fn find_entry_rec(&self, tile_id: u64, entry: &DirEntry, depth: u8) -> Option<DirEntry> {
         // the recursion is done as two functions because it is a bit cleaner,
         // and it allows directory to be cached later without cloning it first.
         let offset = (self.header.leaf_offset + entry.offset) as _;
-        let length = entry.length as _;
-        let dir = self.read_directory(offset, length).await.ok()?;
-        let entry = dir.find_tile_id(tile_id);
 
-        if let Some(entry) = entry {
+        let entry = match self.cache.get_dir_entry(offset, tile_id).await {
+            DirCacheResult::NotCached => {
+                // Cache miss - read from backend
+                let length = entry.length as _;
+                let dir = self.read_directory(offset, length).await.ok()?;
+                let entry = dir.find_tile_id(tile_id).cloned();
+                self.cache.insert_dir(offset, dir).await;
```

Comment:
Do we want to block on this? Not ideal to call `tokio::spawn` here, either, though…

Reply:
the alternative would be a bit worse - we would have to put this future into some var, and then do a `join!(optional_insert, recursive_call)`, and handle a case when insert wasn't needed...

Reply:
plus i suspect NoCache variant will speed up better if it has less of that magic?

Reply:
besides, this could be a subsequent speed up once we have some benchmarks for this